front/lib/resources/storage/models/data_source.ts

```diff
@@ -92,7 +87,7 @@ DataSourceModel.init(
     modelName: "data_source",
     sequelize: frontSequelize,
     indexes: [
-      { fields: ["workspaceId", "name"], unique: true },
+      { fields: ["workspaceId", "name", "deletedAt"], unique: true },
```

Comment:
New index to allow data source names to remain accessible while the soft-deleted ones are being removed.

Reply:
Having deletedAt in the unique here feels a bit weird... I presume it's to allow creating a datasouce after soft deleting it while it's not yet destroyed?

Reply:
Yes like the comment explains...